### PR TITLE
#1355 - Fix PHP Error: Fix grammaticle group label text

### LINKED ISSUE #1355: Group Type Labels - Poor English for You're a vs. You're an
When creating a group type label for any label that should be preceded by "a," groups will display "You're an" instead of "You're a."

Shohel provide the fix that needs to be added to platform ....

`/**
* Add below code to theme/functions.php
*/

add_filter('bp_group_organizer_label_text', 'group_btn_label_text_make_grammatical', 10, 3);
if ( ! function_exists( 'group_btn_label_text_make_grammatical' )) {
	function group_btn_label_text_make_grammatical( $full_text, $group_id, $label_text ) {
		$is_an = preg_match( "/^[aeiou]/i", trim( strtolower( $label_text ) ) );

		if ( $is_an ) {
			$full_text = apply_filters( 'group_btn_label_text_make_grammatical',
				sprintf( __( "You're an %s", 'buddyboss' ), $label_text ), $group_id, $label_text );
		} else {
			$full_text = apply_filters( 'group_btn_label_text_make_grammatical',
				sprintf( __( "You're a %s", 'buddyboss' ), $label_text ), $group_id, $label_text );
		}

		return $full_text;
	}
}`

### PR DESCRIPTION
**Pull request by kishanjasani**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #1355

### How to test the changes in this Pull Request:

1. When creating a group type label for any label that should be preceded by "a," groups will display "You're an" instead of "You're a."
2. Go to your -> site/groups/
3. See label

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: src/bp-groups/bp-groups-template.php [hunk 1]
```diff
@@ -4368,6 +4368,28 @@ function bp_get_group_join_button( $group = false ) {
...
 
+if ( ! function_exists( 'bb_group_btn_label_text_make_grammatical' ) ) {
+	/**
+	 * Make Correct grammaticle group label.
+	 *
+	 * @param string $full_text  Label of Group User type.
+	 * @param int    $group_id   Group Id
+	 * @param string $label_text Label of Group User type
+	 * @return string
+	 */
```
Comment (chetansatasiya) [inline src/bp-groups/bp-groups-template.php:4379]: @kishanjasani add since tag in doc block.

File: src/bp-groups/bp-groups-template.php [hunk 2]
```diff
@@ -4368,6 +4368,28 @@ function bp_get_group_join_button( $group = false ) {
...
+	function bb_group_btn_label_text_make_grammatical( $full_text, $group_id, $label_text ) {
+		$is_an = preg_match( "/^[aeiou]/i", trim( strtolower( $label_text ) ) );
```
Comment (chetansatasiya) [inline src/bp-groups/bp-groups-template.php:4381]:
```suggestion
		$is_an = apply_filters( 'bb_group_btn_label_text_make_grammatical_regex', preg_match( "/^[aeiou]/i", trim( strtolower( $label_text ) ) ) );
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@kishanjasani Need changes

File: src/bp-groups/bp-groups-template.php (hunk 2 above)
Comment (kishanjasani) [reply src/bp-groups/bp-groups-template.php]: Added a filter.

File: src/bp-groups/bp-groups-template.php (hunk 1 above)
Comment (kishanjasani) [reply src/bp-groups/bp-groups-template.php]: Add since tag in doc comment.

**Review by dipskakadiya — APPROVED** [review]

File: src/bp-groups/bp-groups-template.php [hunk 3]
```diff
@@ -4368,6 +4368,30 @@ function bp_get_group_join_button( $group = false ) {
...
+	function bb_group_btn_label_text_make_grammatical( $full_text, $group_id, $label_text ) {
+		$is_an = apply_filters( 'bb_group_btn_label_text_make_grammatical_regex', preg_match( "/^[aeiou]/i", trim( strtolower( $label_text ) ) ) );
+
+		if ( $is_an ) {
+			$full_text = apply_filters( 'bb_group_btn_label_text_make_grammatical', sprintf( __( "You're an %s", 'buddyboss' ), $label_text ), $group_id, $label_text );
+		} else {
+			$full_text = apply_filters( 'bb_group_btn_label_text_make_grammatical', sprintf( __( "You're a %s", 'buddyboss' ), $label_text ), $group_id, $label_text );
+		}
+
+		return $full_text;
```
Comment (dipskakadiya) [inline src/bp-groups/bp-groups-template.php:4391]: @meetmakadiya apply_filters should be added to this line. why we adding that in if...else.

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

**Comment by KartikSuthar** [comment]
Closing PR as duplicated one already fixed on : https://github.com/buddyboss/buddyboss-platform/pull/2846

**Pull request closed without merging**